var/spack/repos/builtin/packages/fms/package.py

```diff
@@ -80,13 +85,14 @@ class Fms(CMakePackage):
 
     def cmake_args(self):
         args = [
-            self.define_from_variant("64BIT"),
             self.define_from_variant("GFS_PHYS"),
             self.define_from_variant("OPENMP"),
             self.define_from_variant("ENABLE_QUAD_PRECISION", "quad_precision"),
             self.define_from_variant("WITH_YAML", "yaml"),
             self.define_from_variant("CONSTANTS"),
             self.define_from_variant("FPIC"),
+            "-D32BIT:BOOL={}".format("ON" if "precision=32" in self.spec else "OFF"),
+            "-D64BIT:BOOL={}".format("ON" if "precision=64" in self.spec else "OFF"),
```

Comment:
A minor suggestion (beware of typos, since I wrote on the UI only):
```suggestion
            self.define("32BIT", "precision=32" in self.spec),
            self.define("64BIT", "precision=64" in self.spec),
```

Reply:
@alalazo I'll try this out tonight (my home laptop is where I'm doing the testing). I tried something similar last night, but it didn't seem to work, but your code does look better than what I tried!

Reply:
@alalazo Seems to work. I committed it (forgot I could use GitHub to do so).